hypothesis-python/docs/reference/schema_metadata.json

```diff
@@ -0,0 +1,95 @@
+                        "description": "The type of choice made. Corresponds to a call to |PrimitiveProvider.draw_integer|, |PrimitiveProvider.draw_float|, |PrimitiveProvider.draw_string|, |PrimitiveProvider.draw_bytes|, or |PrimitiveProvider.draw_boolean|."
+                    },
+                    "value": {
+                        "description": "The value of the choice. Corresponds to the value returned by a ``PrimitiveProvider.draw_*`` method.\n\n``NaN`` float values are returned as ``[\"float\", <float64_int_value>]``, to distinguish ``NaN`` floats with nonstandard bit patterns. Integers with  ``abs(value) >= 2**63`` are returned as ``[\"integer\", str(value)]``, for compatibility with tools with integer size limitations. Bytes are returned as ``[\"bytes\", base64.b64encode(value)]``."
+                    },
+                    "constraints": {
+                        "type": "object",
+                        "description": "The constraints for this choice. Corresponds to the constraints passed to a ``PrimitiveProvider.draw_*`` method. ``NaN`` float values, integers with ``abs(value) >= 2**63``, and byte values for constraints are transformed as for the ``value`` attribute."
+                    },
+                    "was_forced": {
+                        "type": "boolean",
+                        "description": "Whether this choice was forced. As an implementation detail, Hypothesis occasionally requires that some choices take on a specific value, for instance to end generation of collection elements early for performance. These values are called \"forced\", and have ``was_forced = True``."
+                    }
+                },
+                "required": ["type", "value", "constraints", "was_forced"],
+                "additionalProperties": false
+            }
+        },
+        "choice_spans": {
+            "type": "array",
+            "items": {"type": "array"},
+            "description": ".. warning::\n\n  EXPERIMENTAL AND UNSTABLE. This attribute may change format or disappear without warning.\n\nThe semantically-meaningful spans of the choice sequence of this test case.\n\nEach span has the format ``[label, start, end, discarded]``, where:\n\n* ``label`` is an opaque integer-value string shared by all spans drawn from a particular strategy.\n* ``start`` and ``end`` are indices into the choice sequence for this span, such that ``choices[start:end]`` are the corresponding choices.\n* ``discarded`` is a boolean indicating whether this span was discarded (see |PrimitiveProvider.span_end|).\n\nOnly present if |OBSERVABILITY_CHOICES| is ``True``.\n\n.. note::\n\n  Spans are a relatively low-level implementation detail of Hypothesis, and are exposed in observability for users building tools or research on top of Hypothesis. See |PrimitiveProvider| (and particularly |PrimitiveProvider.span_start| and |PrimitiveProvider.span_end|) for more details about spans."
```

Comment:
We should also say explicitly that the label -> strategy mapping may be different in each test case; it's just "the `n`th strategy".

Reply:
really, it can be different? is that only in edge cases or dynamic construction? it's pretty hard to do any kind of research on top of spans if they aren't stable across test cases

Reply:
Oops, no, when I dig through the code I was thinking of something else.  In general they _can_ vary between runs ([e.g. `calc_label_from_hash`](https://github.com/HypothesisWorks/hypothesis/blob/80669745b130e2fd8dd1ce09d0761d40fa499432/hypothesis-python/src/hypothesis/internal/conjecture/utils.py#L32-L42)), [see also `calc_label`](https://github.com/HypothesisWorks/hypothesis/blob/80669745b130e2fd8dd1ce09d0761d40fa499432/hypothesis-python/src/hypothesis/strategies/_internal/strategies.py#L579-L626), but should be stable between test cases for a single process.

Which probably still needs an explanatory note!

Reply:
let me know if you still want the new text to be changed; I don't really want to say "labels may change between test cases" because that's not really true, it's only for dynamically-constructed strategies like changing what you pass to sampled_from, at which point I would call that a distinct strategy